kube/pod.go

```diff
@@ -204,9 +204,19 @@ func getEnvVars(role *model.Role, defaults map[string]string, secrets SecretRefM
 			continue
 		}
 
-		ok, stringifiedValue := configValue(config, defaults)
-		if !ok {
-			continue
+		var stringifiedValue string
+		if createHelmChart {
+			required := ""
+			if config.Required {
+				required = fmt.Sprintf(`required "A valid .Values.env.%s is required" `, config.Name)
```

Comment:
It is just following the example from the https://docs.helm.sh/developing_charts/#charts_tips_and_tricks documentation:

```
value: {{required "A valid .Values.who entry required!" .Values.who }}
```

I find the whole concept weird that both the `required` function as well as the `default` function are used during template expansion instead of being attributes in the `values.yaml` file, but I guess that is a separate topic.

The `.Values.env` part is included because that is the full scope of the variables defined in `values.yaml` right now.  If you use them in a template they would be `{{ .Values.env.UAA_PORT }}`.